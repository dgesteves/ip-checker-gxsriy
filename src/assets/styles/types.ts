import "styled-components";

declare module "styled-components" {
  export interface DefaultTheme {
    colors: IColors;
    sizes: ISizes;
    breakpoints: IBreakpoints;
  }
}

export interface IColors {
  white: "#FFFFFF";
  very_dark_gray: "hsl(0, 0%, 17%)";
  dark_gray: "hsl(0, 0%, 59%)";
  gray: "hsl(0, 0%, 80%)";
  light_gray: "hsl(0, 0%, 98%)";
  error: "hsl(0, 100%, 50%)";
}

export interface ISizes {
  s2: "0.125rem";
  s4: "0.25rem";
  s6: "0.375rem";
  s8: "0.5rem";
  s10: "0.625rem";
  s12: "0.75rem";
  s14: "0.875rem";
  s16: "1rem";
  s18: "1.125rem";
  s20: "1.25rem";
  s24: "1.5rem";
  s28: "1.75rem";
  s32: "2rem";
  s36: "2.25rem";
  s40: "2.5rem";
  s42: "2.625rem";
  s44: "2.75rem";
  s48: "3rem";
  s52: "3.25rem";
  s56: "3.5rem";
  s60: "3.75rem";
  s64: "4rem";
  s68: "4.25rem";
  s72: "4.5rem";
  s76: "4.75rem";
  s80: "5rem";
  s90: "5.5rem";
  s100: "6rem";
}

export interface IBreakpoints {
  mobile: "375px";
  desktop: "1440px";
}
