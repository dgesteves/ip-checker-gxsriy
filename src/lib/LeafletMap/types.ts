export default interface ILeafletMapProps {
  lat: number;
  lng: number;
}
