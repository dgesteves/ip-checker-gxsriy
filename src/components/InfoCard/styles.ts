import styled from "styled-components";

export const StyledInfoCard = styled.section`
  position: absolute;
  bottom: -15%;
  z-index: 1000;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: ${({ theme }) => theme.colors.white};
  width: ${({ theme }) => `calc(${theme.sizes.s100}*10)`};
  height: ${({ theme }) => theme.sizes.s100};
  border-radius: ${({ theme }) => theme.sizes.s8};
  padding: ${({ theme }) => theme.sizes.s24};
  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    flex-direction: column;
    width: ${({ theme }) => `calc(${theme.sizes.s100}*3)`};
    height: ${({ theme }) => `calc(${theme.sizes.s100}*3)`};
    bottom: -55%;
  }
`;

export const StyledInfoCardElement = styled.div`
  display: flex;
  flex-direction: column;
  gap: ${({ theme }) => theme.sizes.s8};

  :not(:first-child) {
    border-left: 1px solid ${({ theme }) => theme.colors.gray};
    padding-left: ${({ theme }) => theme.sizes.s18};
  }

  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    align-items: center;
    :not(:first-child) {
      border-left: none;
      padding-left: 0;
    }
  }
`;

export const StyledInfoCardElementTitle = styled.h4`
  font-size: ${({ theme }) => theme.sizes.s12};
  font-weight: 500;
  text-transform: uppercase;
`;

export const StyledInfoCardElementValue = styled.h3`
  color: ${({ theme }) => theme.colors.very_dark_gray};
  font-size: ${({ theme }) => theme.sizes.s18};
  text-align: center;
`;
