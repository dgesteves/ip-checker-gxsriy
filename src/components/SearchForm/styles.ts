import styled from "styled-components";

export const StyledForm = styled.form`
  display: flex;
  justify-content: center;
  align-items: center;
  width: ${({ theme }) => `calc(${theme.sizes.s100}*5)`};
  height: ${({ theme }) => theme.sizes.s44};
  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    width: ${({ theme }) => `calc(${theme.sizes.s100}*3)`};
    touch-action: manipulation;
  }
`;

export const StyledInput = styled.input`
  width: 100%;
  height: 100%;
  border: none;
  border-top-left-radius: ${({ theme }) => theme.sizes.s8};
  border-bottom-left-radius: ${({ theme }) => theme.sizes.s8};
  padding: ${({ theme }) => theme.sizes.s16};
  font-size: ${({ theme }) => theme.sizes.s18};
  outline: none;
`;

export const StyledButton = styled.button`
  height: 100%;
  width: ${({ theme }) => theme.sizes.s44};
  border: none;
  outline: none;
  border-top-right-radius: ${({ theme }) => theme.sizes.s8};
  border-bottom-right-radius: ${({ theme }) => theme.sizes.s8};
  background-color: ${({ theme }) => theme.colors.very_dark_gray};
  cursor: pointer;
  transition: all 0.4s ease-in-out;

  :hover {
    width: ${({ theme }) => theme.sizes.s64};
  }
`;

export const StyledErrorMessage = styled.p`
  color: ${({ theme }) => theme.colors.error};
  font-size: ${({ theme }) => theme.sizes.s18};
  font-weight: 500;
  text-shadow: 0 0 80px rgb(192 219 255 / 35%), 0 0 32px rgb(65 120 255 / 24%);
`;
