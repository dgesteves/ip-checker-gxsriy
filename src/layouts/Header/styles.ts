import styled from "styled-components";

export const StyledHeader = styled.header<{ img: string }>`
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: ${({ theme }) => theme.sizes.s24};
  height: 30vh;
  padding: ${({ theme }) => theme.sizes.s24};
  background-image: url(${({ img }) => img});
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
`;

export const StyledHeaderTitle = styled.h1`
  color: ${({ theme }) => theme.colors.white};
  font-size: ${({ theme }) => theme.sizes.s28};
  font-weight: 500;
}
  `;
