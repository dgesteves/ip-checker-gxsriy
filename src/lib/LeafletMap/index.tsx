import { TileLayer, Marker } from "react-leaflet";
import { StyledMapContainer } from "./styles";
import ILeafletMapProps from "./types";
import costumeMarker from "../costumeMarker";
import RecenterView from "../RecenterView";
import { MAP_TILE_LAYER, MAP_ZOOM } from "../../constants/app";

export default function LeafletMap({ lat, lng }: ILeafletMapProps) {
  return (
    <section role="application">
      <StyledMapContainer
        center={[lat, lng]}
        zoom={MAP_ZOOM}
        scrollWheelZoom={false}
        zoomControl={false}
      >
        <RecenterView center={[lat, lng]} />
        <TileLayer
          attribution={MAP_TILE_LAYER.ATTRIBUTION}
          url={MAP_TILE_LAYER.URL}
        />
        <Marker position={[lat, lng]} icon={costumeMarker} />
      </StyledMapContainer>
    </section>
  );
}
