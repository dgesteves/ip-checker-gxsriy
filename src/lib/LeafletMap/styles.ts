import styled from "styled-components";
import { MapContainer } from "react-leaflet";

export const StyledMapContainer = styled(MapContainer)`
  height: 70vh;
`;
