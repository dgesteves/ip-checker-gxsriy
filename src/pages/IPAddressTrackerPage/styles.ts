import styled from "styled-components";

export const StyledMapFallback = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 70vh;
`;

export const StyledMapFallbackText = styled.p`
  background-color: ${({ theme }) => theme.colors.error};
  color: ${({ theme }) => theme.colors.white};
  padding: ${({ theme }) => theme.sizes.s8} ${({ theme }) => theme.sizes.s16};
  border-radius: ${({ theme }) => theme.sizes.s8};
  font-size: ${({ theme }) => theme.sizes.s24};
  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    width: ${({ theme }) => `calc(${theme.sizes.s100}*3)`};
  }
`;
