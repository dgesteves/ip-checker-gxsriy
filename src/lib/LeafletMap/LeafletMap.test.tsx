import { render } from "@testing-library/react";
import { theme } from "../../assets/styles/theme";
import { ThemeProvider } from "styled-components";
import GlobalStyles from "../../assets/styles/globalStyles";
import LeafletMap from "./index";
import { GETIPDATA_RESPONSE } from "../../constants/tests";

describe("LeafletMap component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <LeafletMap
          lat={GETIPDATA_RESPONSE.location.lat}
          lng={GETIPDATA_RESPONSE.location.lng}
        />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <LeafletMap
          lat={GETIPDATA_RESPONSE.location.lat}
          lng={GETIPDATA_RESPONSE.location.lng}
        />
      </ThemeProvider>
    );

    expect(container).toBeInTheDocument();
  });
});
