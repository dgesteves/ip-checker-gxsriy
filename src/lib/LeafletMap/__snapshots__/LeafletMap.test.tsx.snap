// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeafletMap component renders correctly 1`] = `
<div>
  <section
    role="application"
  >
    <div
      class="sc-bczRLJ eWDuRx leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
      style="position: relative;"
      tabindex="0"
    >
      <div
        class="leaflet-pane leaflet-map-pane"
        style="left: 0px; top: 0px;"
      >
        <div
          class="leaflet-pane leaflet-tile-pane"
        >
          <div
            class="leaflet-layer "
            style="z-index: 1;"
          >
            <div
              class="leaflet-tile-container leaflet-zoom-animated"
              style="z-index: 18; left: 0px; top: 0px;"
            >
              <img
                alt=""
                class="leaflet-tile"
                role="presentation"
                src="https://c.tile.openstreetmap.org/17/116424/51608.png"
                style="width: 256px; height: 256px; left: -103px; top: -180px;"
              />
            </div>
          </div>
        </div>
        <div
          class="leaflet-pane leaflet-overlay-pane"
        />
        <div
          class="leaflet-pane leaflet-shadow-pane"
        />
        <div
          class="leaflet-pane leaflet-marker-pane"
        >
          <img
            alt="Marker"
            class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
            role="button"
            src="icon-location.svg"
            style="left: 0px; top: 0px; z-index: 0;"
            tabindex="0"
          />
        </div>
        <div
          class="leaflet-pane leaflet-tooltip-pane"
        />
        <div
          class="leaflet-pane leaflet-popup-pane"
        />
      </div>
      <div
        class="leaflet-control-container"
      >
        <div
          class="leaflet-top leaflet-left"
        />
        <div
          class="leaflet-top leaflet-right"
        />
        <div
          class="leaflet-bottom leaflet-left"
        />
        <div
          class="leaflet-bottom leaflet-right"
        >
          <div
            class="leaflet-control-attribution leaflet-control"
          >
            <a
              href="https://leafletjs.com"
              title="A JavaScript library for interactive maps"
            >
              Leaflet
            </a>
             
            <span
              aria-hidden="true"
            >
              |
            </span>
             Â© 
            <a
              href="https://www.openstreetmap.org/copyright"
            >
              OpenStreetMap
            </a>
             contributors
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
